app/models/hyrax/collection_type.rb

```diff
@@ -110,17 +108,33 @@ def ensure_no_collections
         throw :abort
       end
 
-      def ensure_no_settings_changes
+      def ensure_no_settings_changes_for_admin_set_type
+        return true unless admin_set? && exists_for_machine_id?(ADMIN_SET_MACHINE_ID)
```

Comment:
I don't quite understand what is being checked here.  Is this the same as `unless admin_set? && persisted?`?  Why do you need to check for existence because modifying admin_set settings is always forbidden, right?

If you don't need to check existence I think these three cases could be collapsed together.